src/charm.py

```diff
@@ -176,14 +189,33 @@ def _reconcile(self) -> None:
                     self._ingress_provider
                 )
                 tls_information = TLSInformation.from_charm(self, self.certificates)
+                self.unit.set_ports(80, 443)
                 self.haproxy_service.reconcile_ingress(
                     config, ingress_requirers_information, tls_information.external_hostname
                 )
             case ProxyMode.LEGACY:
+                legacy_invalid_requested_port: list[str] = []
+                required_ports: set[Port] = set()
+                for service in self.reverseproxy_requirer.get_services_definition().values():
+                    port = service["service_port"]
+                    if not _validate_port(port):
+                        logger.error("Requested port: %s is not a valid tcp port. Skipping", port)
+                        legacy_invalid_requested_port.append(f"{service['service_name']:{port}}")
+                        continue
+                    required_ports.add(Port(protocol="tcp", port=port))
+
+                if legacy_invalid_requested_port:
```

Comment:
I saw the discussion in the previous comment but I wonder if this port was passed via relation, shouldn't it emit some kind of event to the require warning about it instead of blocking the charm?
Example:
https://github.com/canonical/prometheus-k8s-operator/blob/62cd2d9b8c3a528910ce12c553209817e1b7a889/lib/charms/prometheus_k8s/v1/prometheus_remote_write.py#L114

Reply:
There's currently no standardised interface for the `reverseproxy` relation and afaik the charms using this legacy interface is quite old and probably won't be updated in the future. So I think it's better to block in on our side since even if we pass the information over the relation data there's no guarantee that the other charm can handle it